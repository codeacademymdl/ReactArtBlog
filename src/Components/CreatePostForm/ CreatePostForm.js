import React, { Component } from 'react';
import './CreatePostForm.css';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';


class CreatePostForm extends Component {

	render() {
		return (

			<form className="form-width">
				<ul className="flex-outer">
					<li>
						<label for="date-form">Date</label>
						<input type="date" id="date" />
					</li>
					<li>
						<label for="read-time">Read Time</label>
						<input type="text" id="read-time" placeholder="Enter the read time for blog" />
					</li>
					<li>
						<label for="title">Title</label>
						<input type="text" id="title" placeholder="Enter the Title of Post" />
					</li>
					<li>
						<label for="message">Message</label>
						<textarea rows="6" id="message" placeholder="Enter your message here"></textarea>
					</li>
					<li>
						<Router>
						<Link to={`/`}>
							<button type="submit">Submit</button>
						</Link>

						</Router>

					</li>
				</ul>
			</form>


		);
	}
}








export default CreatePostForm;



