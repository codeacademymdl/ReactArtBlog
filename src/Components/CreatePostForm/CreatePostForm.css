.form-width {
    width: 50%;
    margin: auto;
  }
  .flex-outer li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: 5%;
    margin-bottom: 5%;
  }
  .flex-outer > li > label,
  .flex-outer li p {
    flex: 1 0 120px;
    max-width: 220px;
  }
   
  .flex-outer > li > label + *{
    flex: 1 0 220px;
  }
  .flex-outer li button {
      margin-left: auto;
      margin-right: auto;
      margin-top: 20px;
      padding: 8px 16px;
      border: none;
      background: #333;
      color: #f2f2f2;
      text-transform: uppercase;
      letter-spacing: .09em;
      border-radius: 8px;
    }