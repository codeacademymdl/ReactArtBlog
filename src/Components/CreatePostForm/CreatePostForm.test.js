import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CreatePostForm from './ CreatePostForm'

describe('CreatePostForm', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<CreatePostForm/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});

