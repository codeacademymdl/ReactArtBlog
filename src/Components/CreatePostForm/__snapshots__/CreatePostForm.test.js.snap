// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePostForm renders correctly 1`] = `
<form
  className="form-width"
>
  <ul
    className="flex-outer"
  >
    <li>
      <label
        for="date-form"
      >
        Date
      </label>
      <input
        id="date"
        type="date"
      />
    </li>
    <li>
      <label
        for="read-time"
      >
        Read Time
      </label>
      <input
        id="read-time"
        placeholder="Enter the read time for blog"
        type="text"
      />
    </li>
    <li>
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="title"
        placeholder="Enter the Title of Post"
        type="text"
      />
    </li>
    <li>
      <label
        for="message"
      >
        Message
      </label>
      <textarea
        id="message"
        placeholder="Enter your message here"
        rows="6"
      />
    </li>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          type="submit"
        >
          Submit
        </button>
      </a>
    </li>
  </ul>
</form>
`;
